### ISSUE TITLE
Please use semantic versioning

### ISSUE THREAD

**Issue by rvanlaar** [jazzband/django-axes#252]
Hello,

We're an avid user of Django Axes. We use it to guard a custom user model. 

Therefore we've implemented Django Axes a bit differently than normal.
We were using the api in the decorators.py file.

Between 2.3.2 and 2.3.3 the external api didn't change. However the internal API 
changed significantly. Queue to us having complains from users that they can't login anymore.
Specifically the way the cache key was set up tripped us over.

So to go forward: 
- Please include a more detailed changelog
- Please switch to semantic versioning with this needing a major version bump.

This seems related:
https://github.com/jazzband/django-axes/issues/246

Regards,

Roland

tldr: Use semantic versioning.

**Comment by camilonova** [comment]
Thank you.

Since this is a community driven project we should have more eyes to make sure those kind of issues doesn't affect this library.

Feel free to participate and enforce to have a more detailed changelog and proper use of semantic versioning (which we already use)
